test/unit/metrics/test_metricframe_aggregates.py

```diff
@@ -1040,3 +1040,195 @@ def test_2m_1sf_2cf():
         assert ratios_overall[mfn][('m', 'aa')] == min(ratio_overall_m_a)
         assert ratios_overall[mfn][('m', 'ba')] == pytest.approx(min(ratio_overall_m_b),
                                                                  rel=1e-10, abs=1e-16)
+
+
+class Test2m1sf1cfErrorHandlingCM:
+    # Metricframe containing a confusion matrix raises ValueErrors, since cells are non-scalar
+    def _prepare(self):
+        fns = {'recall': skm.recall_score, 'confusion_matrix': skm.confusion_matrix}
+        self.target = metrics.MetricFrame(metrics=fns, y_true=y_t, y_pred=y_p,
+                                          sensitive_features=g_2,
+                                          control_features=g_3)
+
+        self.expected = metrics.MetricFrame(metrics={'recall': skm.recall_score}, y_true=y_t, y_pred=y_p,
```

Comment:
If this is being used how I think it is, then what you're doing is _dropping_ any columns which don't respond well to `min()`, rather than coercing to NaN?

Reply:
Aj yes you are indeed right. The behaviour now is that it will ignore columns that are non-scalar, but it indeed has to output NaN. Will edit this tomorrow.